starter-core/src/main/java/io/micronaut/starter/feature/ci/workflows/github/GithubCiWorkflowFeature.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.ci.workflows.github;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.feature.ci.workflows.CIWorkflowFeature;
+import io.micronaut.starter.feature.ci.workflows.github.templates.javaAction;
+import io.micronaut.starter.options.JdkDistribution;
+import io.micronaut.starter.template.RockerTemplate;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class GithubCiWorkflowFeature extends CIWorkflowFeature {
+    public static final String NAME = "github-workflow-ci";
+
+    @Override
+    @NonNull
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public String getTitle() {
+        return "Github Actions CI Workflow";
+    }
+
+    @Override
+    @NonNull
+    public String getDescription() {
+        return "Adds a GitHub Actions Workflow to build a Micronaut application";
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        super.apply(generatorContext);
+
+        String workflowFilePath = ".github/workflows/" + getWorkflowFileName(generatorContext);
+        generatorContext.addTemplate("javaAction",
+                new RockerTemplate(workflowFilePath,
+                        javaAction.template(
```

Comment:
extract to method. e.g. `RockerModel workflowFileRockerModel(GeneratorContext generatorContext)`